docker/convert.go

```diff
@@ -114,7 +117,7 @@ func ports(c *config.ServiceConfig) (map[nat.Port]struct{}, nat.PortMap, error)
 }
 
 // Convert converts a service configuration to an docker API structures (Config and HostConfig)
-func Convert(c *config.ServiceConfig, ctx project.Context) (*container.Config, *container.HostConfig, error) {
+func Convert(c *config.ServiceConfig, ctx project.Context, clientFactory composeclient.Factory) (*container.Config, *container.HostConfig, error) {
```

Comment:
The `client.Factory` is there for handling `container:xxxx` network modes (and in fact, just get the container ID :smile:). Handling networkMode should be done elsewhere (but I ran out of idea in this PR :smile_cat: ).